src/main/scala/viper/gobra/translator/implementations/components/OptionToSeqImpl.scala

```diff
@@ -0,0 +1,125 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2020 ETH Zurich.
+
+package viper.gobra.translator.implementations.components
+
+import viper.gobra.translator.interfaces.Collector
+import viper.gobra.translator.interfaces.components.{OptionToSeq, Options}
+import viper.silver.ast.{DomainFuncApp, ErrorTrafo, Exp, Info, Position}
+import viper.silver.{ast => vpr}
+
+class OptionToSeqImpl(options : Options) extends OptionToSeq {
+  private val domainName : String = "OptToSeq"
+  private lazy val domainFuncName : String = "opt2seq"
+  private val typeVar : vpr.TypeVar = vpr.TypeVar("T")
+
+  /**
+    * Determines whether the Viper domain
+    * should be generated upon finalisation.
+    */
+  private var generateDomain : Boolean = false
+
+  /**
+    * Definition of the "opt2seq" domain function:
+    *
+    * {{{
+    * function opt2seq(o : Option[T]) : Seq[T]
+    * }}}
+    */
+  private lazy val domainFunc : vpr.DomainFunc = vpr.DomainFunc(
+    domainFuncName,
+    Seq(vpr.LocalVarDecl("o", options.typ(typeVar))()),
+    vpr.SeqType(typeVar)
+  )(domainName = domainName)
+
+  /**
+    * Definition of the "opt2seq_none" axiom:
+    *
+    * {{{
+    * axiom opt2seq_none {
+    *   forall o : Option[T] :: opt2seq(o) == Seq() <==> o == optnone()
```

Comment:
The axiomatization seems weird to me. I would expect something along the lines of:
`opt2seq(optnone()) == Seq()  && forall e :: opt2seq(optsome(e)) == Seq(e)`.
A separate axiom should specify that an option is either some or none. I think we had some previous work on how ADT are encoded into Viper.

Reply:
I think that we usually use some "tag" utility function to distinguish between the elements.

Reply:
I updated the axiom according to your suggestion. Regarding "tag" functions; do you have an example or pointer?

Reply:
I will send you a mail with two sources in a minute.